### PR TITLE
Feature/rx disposable

### PR DESCRIPTION
**Pull request by jopmiddelkamp**
I've added a reactive disposable mixin to make it easier to dispose of stream listeners. With this mixin, there is no need anymore to store and close stream subscriptions for simple stream listeners.

### CODE REVIEW

**Comment by jopmiddelkamp** [comment]
> # [Codecov](https://codecov.io/gh/hoc081098/rxdart_ext/pull/52?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Petrus+Nguy%E1%BB%85n+Th%C3%A1i+H%E1%BB%8Dc) Report
> > Merging [#52](https://codecov.io/gh/hoc081098/rxdart_ext/pull/52?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Petrus+Nguy%E1%BB%85n+Th%C3%A1i+H%E1%BB%8Dc) ([e1d309b](https://github.com/hoc081098/rxdart_ext/commit/e1d309b68cbf685e264467528c6cefa7509fee6e)) into [master](https://codecov.io/gh/hoc081098/rxdart_ext/commit/d2490c5828cd8b90ae1fd64475488b650d923f43?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Petrus+Nguy%E1%BB%85n+Th%C3%A1i+H%E1%BB%8Dc) ([d2490c5](https://github.com/hoc081098/rxdart_ext/commit/d2490c5828cd8b90ae1fd64475488b650d923f43)) will **decrease** coverage by `0.85%`.
> > The diff coverage is `0.00%`.
> 
> [Impacted file tree graph](https://codecov.io/gh/hoc081098/rxdart_ext/pull/52?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Petrus+Nguy%E1%BB%85n+Th%C3%A1i+H%E1%BB%8Dc)
> 
> ```diff
> @@            Coverage Diff             @@
> ##           master      #52      +/-   ##
> ==========================================
> - Coverage   95.10%   94.25%   -0.86%     
> ==========================================
>   Files          43       44       +1     
>   Lines         552      557       +5     
> ==========================================
>   Hits          525      525              
> - Misses         27       32       +5     
> ```
> 
> [Impacted Files](https://codecov.io/gh/hoc081098/rxdart_ext/pull/52?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Petrus+Nguy%E1%BB%85n+Th%C3%A1i+H%E1%BB%8Dc)	Coverage Δ	
> [lib/src/utils/disposable.dart](https://codecov.io/gh/hoc081098/rxdart_ext/pull/52/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Petrus+Nguy%E1%BB%85n+Th%C3%A1i+H%E1%BB%8Dc#diff-bGliL3NyYy91dGlscy9kaXNwb3NhYmxlLmRhcnQ=)	`0.00% <0.00%> (ø)`	
> [Continue to review full report at Codecov](https://codecov.io/gh/hoc081098/rxdart_ext/pull/52?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Petrus+Nguy%E1%BB%85n+Th%C3%A1i+H%E1%BB%8Dc).
> 
> > **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Petrus+Nguy%E1%BB%85n+Th%C3%A1i+H%E1%BB%8Dc)
> > `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> > Powered by [Codecov](https://codecov.io/gh/hoc081098/rxdart_ext/pull/52?src=pr&el=footer&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Petrus+Nguy%E1%BB%85n+Th%C3%A1i+H%E1%BB%8Dc). Last update [d2490c5...e1d309b](https://codecov.io/gh/hoc081098/rxdart_ext/pull/52?src=pr&el=lastupdated&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Petrus+Nguy%E1%BB%85n+Th%C3%A1i+H%E1%BB%8Dc). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Petrus+Nguy%E1%BB%85n+Th%C3%A1i+H%E1%BB%8Dc).

I'll take a look into this later this week.

**Comment by hoc081098** [comment]
Hi @jopmiddelkamp, do you have any updates on this?

File: test/utils/disposable_test.dart [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  _MockClass(
+    this.completer,
+  ) {
+    dispose$.listen((_) => completer.complete());
+  }
+
+  final Completer<void> completer;
+}
+
+void main() {
```
Comment (hoc081098) [inline test/utils/disposable_test.dart:16]: Please include this test in `rxdart_ext_test.dart` file

Comment (jopmiddelkamp) [reply test/utils/disposable_test.dart:16]: Done

**Review by hoc081098 — APPROVED** [review]
Thanks @jopmiddelkamp 👍

**Pull request merged**